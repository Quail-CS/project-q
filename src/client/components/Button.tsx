/**
 * @module Button.tsx
 * @description Clickable Button
 */

import * as React from 'react';

interface ButtonProps {
  className: string,
  children: string,
  onClick: any, // TODO: Specify
}

const Button = (props: ButtonProps) => {
  const { className, onClick, children } = props
  return (
    <div className={className} onClick={onClick}>
      {children}
    </div>
  )
}

export default Button;
