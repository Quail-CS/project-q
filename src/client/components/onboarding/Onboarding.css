.onboarding {
  grid-row-start: 2; 
  grid-row-end: 5;

  display: flex;
  flex-direction: column;
  align-items: center;

  margin: 4vh 0;
}

.onboarding-step {
  margin: 1vh 0;

  display: flex;
  flex-direction: column; 
  align-items: center;

}

.onboarding-step-text {
  color: #FFFFFF;
  font-size: 1.8rem;
  /* font-weight: bold; */
}

.onboarding-step-icons {
  margin-top: 2vh;
}

.onboarding-icons-img {
  fill: #666666;
  height: 10vh;
  margin: 0 1vw;
}

.onboarding-button {
  color: #888888;
  font-size: 2.4rem;
  font-weight: bold;
  margin-top: 3vh;
}

.onboarding-button:hover {
  color: #FFFFFF;
  cursor: pointer;
}