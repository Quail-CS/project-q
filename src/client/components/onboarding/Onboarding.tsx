/**
 * @module Onboarding.tsx
 * @description Ethiq Onboarding Upon Login
 */

import * as React from 'react';

import './Onboarding.css';

import Button from '../Button';

const doctor = require('../../assets/doctor.png')
const handShake = require('../../assets/hand-shake.png')
const like = require('../../assets/like.png')
const mortarboard = require('../../assets/mortarboard.png')
const rifle = require('../../assets/rifle.png')

interface OnboardingProps {
  updateIssuesSelected: any,
}

const Onboarding = (props: OnboardingProps) => {
  return (
    <div className='onboarding'>
      <div className='onboarding-step'>
        <div className='onboarding-step-text'>Step 1: Select your most important issues.</div>
        <div className='onboarding-step-icons'>
          <img className="onboarding-icons-img" alt="Mortar Board Icon" src={mortarboard} />
          <img className="onboarding-icons-img" alt="Doctor Icon" src={doctor} />
          <img className="onboarding-icons-img" alt="Rifle Icon" src={rifle} />
        </div>
      </div>
      <div className='onboarding-step'>
        <div className='onboarding-step-text'>Step 2: Tell us where you stand on these issues.</div>
        <div className='onboarding-step-icons'>
          
          <img className="onboarding-icons-img" alt="Like Icon" src={like} />
        </div>
      </div>
      <div className='onboarding-step'>
        <div className='onboarding-step-text'>Step 3: Discover companies most aligned with your values.</div>
        <div className='onboarding-step-icons'>
          
          <img className="onboarding-icons-img" alt="Hand Shake Icon" src={handShake} />
        </div>
      </div>
      <Button 
        className="onboarding-button"
        onClick={props.updateIssuesSelected}
      >
        BEGIN
      </Button>
    </div>
  )
};

export default Onboarding;
