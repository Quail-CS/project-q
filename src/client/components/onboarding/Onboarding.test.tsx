import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import Onboarding from './Onboarding';
import Button from '../Button';

Enzyme.configure({ adapter: new Adapter() });

const mockFn = jest.fn();

describe('Onboarding Component', () => {
  it('matches the snapshot', () => {
    const tree: Renderer.ReactTestRendererJSON = Renderer.create(<Onboarding updateIssuesSelected={mockFn} />).toJSON() as Renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  })
  it('renders a <div> with class onboarding', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Onboarding updateIssuesSelected={mockFn} />);
    expect(onboardingWrapper.hasClass('onboarding')).toEqual(true);
    expect(onboardingWrapper.type()).toEqual('div');
  })
  it('contains 3 <div>s with class onboarding-step', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Onboarding updateIssuesSelected={mockFn} />);
    const onboardingStep = onboardingWrapper.find('.onboarding-step');
    expect(onboardingStep).toHaveLength(3);
    expect(onboardingStep.first().type()).toEqual('div');
  })
  it('contains 3 <div>s with class onboarding-step-text', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Onboarding updateIssuesSelected={mockFn} />);
    const onboardingStepText = onboardingWrapper.find('.onboarding-step-text');
    expect(onboardingStepText).toHaveLength(3);
    expect(onboardingStepText.first().type()).toEqual('div');
  })
  it('contains 3 <div>s with class onboarding-step-icons', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Onboarding updateIssuesSelected={mockFn} />);
    const onboardingIcons = onboardingWrapper.find('.onboarding-step-icons');
    expect(onboardingIcons).toHaveLength(3);
    expect(onboardingIcons.first().type()).toEqual('div');
  })
  it('AND contains 5 <img>s with class onboarding-icons-img', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Onboarding updateIssuesSelected={mockFn} />);
    const onboardingIcon = onboardingWrapper.find('.onboarding-icons-img');
    expect(onboardingIcon).toHaveLength(5);
    expect(onboardingIcon.first().type()).toEqual('img');
  })
  it('contains a <Button> with text "Begin"', () => {
    const onboardingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Onboarding updateIssuesSelected={mockFn} />);
    const onboardingButton = onboardingWrapper.find(Button);
    expect(onboardingButton).toHaveLength(1);
    expect(onboardingButton.props().children).toEqual('BEGIN');
  })
});
