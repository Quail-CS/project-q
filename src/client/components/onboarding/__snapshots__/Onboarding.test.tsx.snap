// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Component matches the snapshot 1`] = `
<div
  className="onboarding"
>
  <div
    className="onboarding-step"
  >
    <div
      className="onboarding-step-text"
    >
      Step 1: Select your most important issues.
    </div>
    <div
      className="onboarding-step-icons"
    >
      <img
        alt="Mortar Board Icon"
        className="onboarding-icons-img"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <img
        alt="Doctor Icon"
        className="onboarding-icons-img"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <img
        alt="Rifle Icon"
        className="onboarding-icons-img"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
  </div>
  <div
    className="onboarding-step"
  >
    <div
      className="onboarding-step-text"
    >
      Step 2: Tell us where you stand on these issues.
    </div>
    <div
      className="onboarding-step-icons"
    >
      <img
        alt="Like Icon"
        className="onboarding-icons-img"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
  </div>
  <div
    className="onboarding-step"
  >
    <div
      className="onboarding-step-text"
    >
      Step 3: Discover companies most aligned with your values.
    </div>
    <div
      className="onboarding-step-icons"
    >
      <img
        alt="Hand Shake Icon"
        className="onboarding-icons-img"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
  </div>
  <div
    className="onboarding-button"
    onClick={[MockFunction]}
  >
    BEGIN
  </div>
</div>
`;
